### DISCUSSION TITLE
DialogForm

### DISCUSSION THREAD

**Discussion by lsmoker** [jrief/django-formset#130] [Q&A]
I've started experimenting with the DialogForm within a FormCollection. Is this mainly intended to update fields within the collection for processing/saving when the top-level collection is submitted - or is there a way to do processing immediately when the Submit button on the modal is clicked?

**Comment by jrief** [comment]
The `DialogForm` is intended for all kind of `<dialog>`-s. I already adopted it to the dialogs used in the richtext editor. This will be part of version 1.4. One of its new features is the possibility that you can add buttons to forms and collections, just as you would with normal fields. This allows developers to induce the opening and closing of a dialog using any conditional expression in you django-formset.

It also will be used for [steppers](https://github.com/jrief/django-formset/discussions/86) in `FormCollection`-s

I'm still working on this, so please be patient.

**Comment by lsmoker** [comment]
I see - a work in progress. Very nice.

**Reply by jrief** [reply]
Please check version 1.5. There I added [dialog model forms](https://django-formset.fly.dev/dialog-model-forms/). It might be what you need.
